add-ons/fabric8/README.md

```diff
@@ -0,0 +1,50 @@
+# Fabric8 Addon
+
+This addon provides an easy way to install fabric8 on MiniShift. 
+
+Fabric8 provides a complete developer platform for creating projects, editing them and performing full CI / CD for the entire developer lifecycle.
+
+## Setup GitHub OAuth client ID and secret
+
+Fabric8 has GitHub integration letting you browse repositories, create new repositories, edit projects and setup automated CI / CD jobs with webhooks on github. 
+
+This requires an [OAuth application to be setup on your github account](https://developer.github.com/apps/building-integrations/setting-up-and-registering-oauth-apps/registering-oauth-apps/) for fabric8 and you need to obtain the client ID and secret for the OAuth application.
```

Comment:
I am a bit confused, is there not a chicken and egg problem? _GITHUB_OAUTH_CLIENT_ID_ and  _GITHUB_OAUTH_CLIENT_SECRET_ are marked as required. I installed the plugin and got the following when applying it:

     $ minishift addon apply fabric8
     -- Applying addon 'fabric8':Error applying the add-on: The variable(s) GITHUB_OAUTH_CLIENT_ID, GITHUB_OAUTH_CLIENT_SECRET are required by the add-on, but are not defined in the context.

So obviously I need the variables before that. However, you also want me to run `echo http://keycloak-fabric8.$(minishift ip).nip.io/auth/realms/fabric8/broker/github/endpoint` to determine the URI, but this means I need an initial instance of Fabric8 running, right?

How do I bootstrap this? Should I create a new OAuth Application in GitHub using some dummy URLs and then once bootstrapped update them? Also, how does this work? The URL I'll get will be a host only one. Is that ok for GitHub/OAuth?